### DISCUSSION TITLE
Define a Subset of a Scala 3 Enumeration

### DISCUSSION THREAD

**Discussion by pme123** [Iltotore/iron#237] [Q&A] [answered]
Hi,
I want to define a subset of an Enum. Example:
```
enum DocId:
  case ContractA
  case ContractB
  ..

type ContractProduct = DocId :| any.StrictEqual[DocId.ContractB]
```

This gives me the following compile error:
```
type Vorsorgevereinbarung is not a member of object documents.bpmn.printDocument.DocId.
An extension method was tried, but could not be fully constructed:

    io.github.iltotore.iron.autoRefine[A, C](
      documents.bpmn.printDocument.DocId)
```
Is there a way to achieve this or are Scala 3 enums not supported?

**Comment by Iltotore** [comment]
Hi! Sorry for the delay, I was busy.

Looking at the error, I feel a part of the code is missing in your post. If so, can you post the code where the error occurs?

Anyway `any.StrictEqual[DocId.ContractB]` is not supported by Scala 3 which provides `ValueOf` instances only for primitives but it might be possible to support it in Iron 3 when #147 will be fixed.

**Comment by pme123** [comment]
It is actually the whole code, just the error message is from my concrete project. So in this context it would be:

```
type ContractB is not a member of object mypackage.DocId.
An extension method was tried, but could not be fully constructed:

    io.github.iltotore.iron.autoRefine[A, C](mypackage.DocId)
```

**Reply by Iltotore** [reply]
Ok. According to the error, you tried to refine implicitly at compile-time a `DocId` to a `ContractProduct` like passing a `DocId` where a `ContractProduct` is needed. Can you show this code?

Also I'm not sure about the actual usefulness of such code (it might be only the example). In this case:

```scala
enum DocId:
  case ContractA
  case ContractB
  ..

type ContractProduct = DocId :| any.StrictEqual[DocId.ContractB]
```

Couldn't you use `DocId.ContractB` (or `DocId.ContractB.type` if it's an object)?

**Comment by pme123** [comment, marked as answer]
Ah, thanks. `.type` did the trick: `type ContractProduct = DocId :| any.StrictEqual[DocId.ContractB.type]`

For only one you have right, this would be working: `type ContractProduct = DocId.ContractB.type`. But this was only a simplified example. In reality I need a subset, like: `DocId :| any.In[(DocId.ContractB.type, DocId.ContractC.type)]` which now also works!
